tests/server_notices/test_server_notices_manager.py

```diff
@@ -0,0 +1,197 @@
+            }
+        }
+    )
+    def test_update_notice_user_name_when_changed(self):
+        """
+        Tests that existing server notices user name in room is updated when is
+        different from the one in homeserver config.
+        """
+        self._mock_existing_notice_room(
+            self.server_notices_manager, server_notice_config
+        )
+
+        self.get_success(
+            self.server_notices_manager.send_notice(
+                self.user_id, self.notice_message, EventTypes.Message
+            )
+        )
+
+        self.server_notices_manager._room_member_handler.update_membership.assert_called_once()
+        (
+            _,
+            kwargs,
+        ) = self.server_notices_manager._room_member_handler.update_membership.call_args
+
+        assert kwargs["content"] == {
+            "displayname": self.hs.config.servernotices.server_notices_mxid_display_name,
+            "avatar_url": self.hs.config.servernotices.server_notices_mxid_avatar_url,
+        }
+
+    @override_config(
+        {
+            "server_notices": {
+                **server_notice_config,
+                "system_mxid_avatar_url": "test/new-url",
+            }
+        }
+    )
+    def test_update_notice_user_avatar_when_changed(self):
+        """
+        Tests that existing server notices user avatar in room is updated when is
+        different from the one in homeserver config.
+        """
+        self._mock_existing_notice_room(
+            self.server_notices_manager, server_notice_config
+        )
+
+        self.get_success(
+            self.server_notices_manager.send_notice(
+                self.user_id, self.notice_message, EventTypes.Message
+            )
+        )
+
+        self.server_notices_manager._room_member_handler.update_membership.assert_called_once()
+        (
+            _,
+            kwargs,
+        ) = self.server_notices_manager._room_member_handler.update_membership.call_args
+
+        assert kwargs["content"] == {
+            "displayname": self.hs.config.servernotices.server_notices_mxid_display_name,
+            "avatar_url": self.hs.config.servernotices.server_notices_mxid_avatar_url,
+        }
+
+    def test_doesnt_update_notice_user_profile_when_not_changed(self):
+        """
+        Tests that existing server notices profile in room is not updated when is
+        equal to the one in homeserver config.
+        """
+        self._mock_existing_notice_room(
+            self.server_notices_manager, server_notice_config
+        )
+
+        self.get_success(
+            self.server_notices_manager.send_notice(
+                self.user_id, self.notice_message, EventTypes.Message
+            )
+        )
+
+        self.server_notices_manager._room_member_handler.update_membership.assert_not_called()
+
+    def _mock_existing_notice_room(
+        self, server_notices_manager: ServerNoticesManager, server_notice_config: dict
+    ):
```

Comment:
Instead of mocking all this, I think it would be better to make an actual room with `self.user_id` and the service notice user.

You should be able to use `self.helper.create_room_as(...)` and then `self.helper.join(...)` to get the two users into the room.

Reply:
done. I think you wanted tests more oriented to integration that actual unit tests. I found tests for server_notices following such an approach in `tests/rest/admin/test_server_notice` so I moved my tests there.